zephyr/src/supp_api.c

```diff
@@ -158,6 +158,13 @@ int zephyr_supp_connect(const struct device *dev,
 	struct wpa_supplicant *wpa_s;
 	int ret = 0;
 
+	if (!net_if_is_up(net_if_lookup_by_dev(dev))) {
```

Comment:
In case of iface NULL this throws an assert (if enabled), is it okay or should we check for it?

Reply:
I guess dev can never be NULL as we check in the init itself, so, clsoing.